lib/course.js

```diff
@@ -263,12 +260,41 @@ const applyGlobalOptions = (result, opts, relativePath, parserVersion) => ({
 //
 
 
-module.exports = (dir, { Topic }, opts = {}) => {
-  const readme = path.join(dir, common.getReadmeFileName(opts.locale));
-  const relativePath = path.relative(process.cwd(), dir);
-  const { slug } = common.parseDirname(readme);
+module.exports = async (dirOrPath, { Topic }, opts = {}) => {
+  let readme;
+  let relativePath;
+  let slug;
+  let dir;
+  let location;
+
+  const readmeFileName = common.getReadmeFileName(opts.locale);
+  if (common.isGithubUrl(dirOrPath)) {
+    const { repo, token, branch } = opts;
+    const [username, repository] = repo.split('/');
+    const octokit = Octokit({
+      auth: token,
+    });
+    const { data } = await octokit.repos.getContents({
+      owner: username,
+      repo: repository,
+      path: readmeFileName,
+      ref: branch,
+    });
+    readme = Buffer.from(data.content, 'base64').toString();
+    relativePath = dirOrPath;
+    // TODO: Check with the team if this is fine
+    slug = `${repository}-${branch}`;
```

Comment:
@FabianBravoA I am using both the repository name and the branch name to make sure we don't have any duplicate slugs for courses from a single repository. We have courses maintained in branches in [`mass-training`](https://github.com/Laboratoria/mass-training), for example.

Reply:
As we said, we don't need a completely unique slug name, this is just fine for us now as we're uploading courses into cohorts directly